### ISSUE TITLE
Metabase doesn't respect `Database name` property when connecting to MongoDb

### ISSUE THREAD

**Issue by MichaelSL** [metabase/metabase#18208]
**Describe the bug**
I'm connecting to my MongoDb database hosted on DigitalOcean. `Authentication Database` is `admin`. Metabase doesn't respect `Database name` property and uses schema and data from `Authentication Database` database.

**Logs**
No errors in either console or server log. Server reports that Analyze data step is completed successfully (but tables are from `admin` database).

**To Reproduce**
1. Connect to database using SSL with certificate using `Authentication database` different from `Database name`
2. Sync schema
3. Check tables in your database

**Expected behavior**
Tables should be from `Database name` database
**Screenshots**
![image](https://user-images.githubusercontent.com/2205113/135738894-3bc247d4-3c76-4ac5-9d62-ea6fa9833c87.png)

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Running in Docker
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.61 Safari/537.36 Edg/94.0.992.31",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.12+7",
    "java.vendor": "Eclipse Foundation",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.12",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.12+7",
    "os.name": "Linux",
    "os.version": "5.10.16.3-microsoft-standard-WSL2",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-09-21",
      "tag": "v0.40.5",
      "branch": "release-x.40.x",
      "hash": "042a36e"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```
**Severity**
Blocking my usage of Metabase entirely: I can't analyze my prod data.

**Additional context**
I'm able to connect to my local copy of the database. Local MongoDb runs with the same configuration (users in `admin` db) but without SSL.

**Comment by jeff303** [comment]
I'm afraid that I'm unable to reproduce this, at least on our local Mongo 4 test Docker instance.  I did the following steps:

(log into the DB using the `mongo` cli tool, then...)

```
use authdb
db.createUser({user: "jeff",pwd: "mypassword",roles: [ { role: "readWrite", db: "test-data" } ],mechanisms: [ "SCRAM-SHA-256" ]})
```

Note that `test-data` is an already existing DB, containing our test dataset (i.e. categories, checkins, users, venues).  After this, I create a DB connection using `authdb` as the Authentication database, and `test-data` as the Database name.  These four tables sync as expected (there are no actual tables in `authdb` in this case).

**Comment by MichaelSL** [comment]
@jeff303 yep, I have similar setup locally and it works fine. 

The issue exists only when connecting to Digital Ocean Mongo Db cluster. I assume it may be because of SSL cert authentication.

I'll also try to check DO logs, if I'll find anything useful.

**Comment by MichaelSL** [comment]
I see quite a lot of `No SSL certificate provided by peer` messages. Is it enough to set "Use a secure connection (SSL)" toggle and copy the certificate text?
![image](https://user-images.githubusercontent.com/2205113/136446312-cc115571-bcc1-4ca6-8d2d-13d0aacda8ac.png)

**Comment by maksspace1** [comment]
Any news about this one?

**Comment by MichaelSL** [comment]
Unfortunately I'm not able to reproduce locally yet.

- Michael
________________________________

**Comment by spike008t** [comment]
I have the same issue, but I think I've found the issue here the PR #19114

**Comment by jeff303** [comment]
Fixed under #19114.

**Comment by ChristianRuff-QraGo** [comment]
Hi guys,
Is there a workaround available for this issue? 
BR
